Mage.Sets/src/mage/cards/c/ChissGoriaForgeTyrant.java

```diff
@@ -0,0 +1,204 @@
+import mage.abilities.effects.AsThoughEffectImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.cost.CostModificationEffectImpl;
+import mage.abilities.keyword.AffinityForArtifactsAbility;
+import mage.abilities.keyword.FlyingAbility;
+import mage.abilities.keyword.HasteAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.Cards;
+import mage.cards.CardsImpl;
+import mage.constants.*;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.stack.Spell;
+import mage.players.Player;
+import mage.util.CardUtil;
+import mage.watchers.Watcher;
+
+/**
+ * @author Xanderhall
+ */
+public final class ChissGoriaForgeTyrant extends CardImpl {
+    
+    public ChissGoriaForgeTyrant(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{6}{R}{R}{R}");
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.DRAGON);
+
+        this.power = new MageInt(5);
+        this.toughness = new MageInt(4);
+
+        // Affinity for artifacts
+        this.addAbility(new AffinityForArtifactsAbility());
+
+        // Flying, haste
+        this.addAbility(FlyingAbility.getInstance());
+        this.addAbility(HasteAbility.getInstance());
+
+        /*
+         * Whenever Chiss-Goria, Forge Tyrant attacks, exile the top five cards of your library. 
+         * You may cast an artifact spell from among them this turn. If you do, it has affinity for artifacts.
+         */
+
+        this.addAbility(new AttacksTriggeredAbility(new ChissGoriaForgeTyrantEffect()), new ChissGoriaForgeTyrantWatcher());
+    }
+
+    private ChissGoriaForgeTyrant(final ChissGoriaForgeTyrant card) {
+        super(card);
+    }
+
+    @Override
+    public ChissGoriaForgeTyrant copy() {
+        return new ChissGoriaForgeTyrant(this);
+    }
+}
+
+class ChissGoriaForgeTyrantEffect extends OneShotEffect {
+
+    ChissGoriaForgeTyrantEffect() {
+        super(Outcome.Benefit);
+        staticText = "exile the top five cards of your library. You may cast an artifact spell from among them this turn. If you do, it gains affinity for artifacts.";
+    }
+
+    private ChissGoriaForgeTyrantEffect(final ChissGoriaForgeTyrantEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public ChissGoriaForgeTyrantEffect copy() {
+        return new ChissGoriaForgeTyrantEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+
+        // Return if no cards exiled, effect still applied
+        Cards cards = new CardsImpl(player.getLibrary().getTopCards(game, 5));
+        if (cards.isEmpty()) {
+            return true;
+        }
+        player.moveCards(cards, Zone.EXILED, source, game);
```

Comment:
Might want to use `moveCardsToExile` to have an exile zone specific to this effect